datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -702,6 +703,14 @@ struct SMJStream {
     pub reservation: MemoryReservation,
     /// Runtime env
     pub runtime_env: Arc<RuntimeEnv>,
+    pub streamed_batch_counter: AtomicUsize,
+}
+
+struct JoinedRecordBatches {
```

Comment:
we need more information to track the filtered bitmask across incoming batches

Reply:
Could you please comment what the fields represent -- In particular how is `filter_mask`, `row_indices` and `batch_ids`  interpreted relative to the `batches` (do they always have the same row count? What does the batch represent?